pre-analysis/COMMON_dilution.py

```diff
@@ -0,0 +1,291 @@
+                sys.exit(255)
+                   
+
+            if process.instrument.type == "QIAgility":
+                conc = outart.samples[0].udf["Sample concentration (ng/ul)"] / Pre_Dilution_Factor
+
+            else:
+                conc = outart.samples[0].udf["Sample concentration (ng/ul)"] 
+
+            logging.info(f"Found {inart.name} in {well} with concentration {conc:.2f} ng/µl")
+
+            
+            # check for a target concentration, for NPM1 it is always manually choosen
+
+            try:
+                targetVolume = outart.udf["Target volume (ul)"]
+            except KeyError:
+                logging.error("Fail: Choose Target volume.")
+                sys.exit(255)
+
+            targetConc = outart.udf["Target concentration (ng/ul)"]
+
+            if conc < targetConc:
+                sampleVol = targetVolume
+                dilutionFactor = 1
+            else:
+                dilutionFactor = conc / targetConc
+
+            sampleVol = targetVolume / dilutionFactor
+
+            if sampleVol < 2:
+                sampleVol = 2
+                targetVolume = sampleVol * dilutionFactor
+                outart.udf["Target volume (ul)"] = targetVolume
+
+            buffVol = targetVolume - sampleVol
+
+            outart.udf["Dilution factor"] = dilutionFactor
+            outart.udf["Sample volume (ul)"] = sampleVol
+            outart.udf["Buffer volume (ul)"] = buffVol
+            outart.put()
+
+            containers = process.output_containers()
+
+            if containers[0].type.id == container_ids["96wellplate"].id:
+                sort_name = Well(artifact.location[1]).rev_zero() + "_" + artifact.name
+            else:
+#                unsort_name = inart.name
+#                print(unsort_name)
+#                sort_name = sorted(unsort_name, key=lambda item: (''.join(filter(str.isalpha, item)), int(''.join(filter(str.isdigit, item)))))
+                sort_name = inart.name
+
+            sort_name_tuple = tuple(sort_name)
+
+            concentration[sort_name] = {
+                'Sample Name': inart.name,
+                'Clarity LimsID': inart.samples[0].id,
+                'Container': inart.location[0].name,
+                'Well': well,
+                'Concentration type': conType,
+                'Concentration (ng/ul)': conc,
+                'Target concentration (ng/ul)': targetConc,
+                'Target volume (ul)': targetVolume,
+                'Dilution factor': dilutionFactor,
+                'Sample volume (ul)': sampleVol,
+                'Buffer volume (ul)': buffVol,
+                'Analysis': analysis,
+            }
+
+
+    return concentration
+
+
+def create_lines_QIAgility(concentration):
+    """Write one line for each sampleid.
+    Add concentration for each sampleid.
+    """
+
+    logging.info("Creating lines")
+
+    lines = [SEPARATOR["Separator semicolon"].join(HEADER["Header QIAgility"])]
```

Comment:
Instead of writing "Header QIAgility" use the instrument name from Clarity. Add an input to the function which is instrument_name.
(also change SEPARATOR if the dict is changed)
```suggestion
    lines = [SEPARATOR[instrument_name].join(HEADER[instrument_name])]
```